src/main/kotlin/no/nav/syfo/client/pdl/PdlClient.kt

```diff
@@ -4,21 +4,44 @@ import io.ktor.client.call.*
 import io.ktor.client.request.*
 import io.ktor.client.statement.*
 import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
 import no.nav.syfo.client.ClientEnvironment
 import no.nav.syfo.client.azuread.AzureAdClient
 import no.nav.syfo.client.httpClientDefault
 import no.nav.syfo.client.pdl.domain.*
 import no.nav.syfo.domain.PersonIdentNumber
 import no.nav.syfo.util.*
 import org.slf4j.LoggerFactory
+import java.util.concurrent.ConcurrentHashMap
 
 class PdlClient(
     private val azureAdClient: AzureAdClient,
     private val pdlEnvironment: ClientEnvironment,
 ) {
     private val httpClient = httpClientDefault()
+    val cache = ConcurrentHashMap<PersonIdentNumber, PdlHentIdenter>()
```

Comment:
Best å legge en sånn cache i et companion object slik at den blir delt mellom alle instanser av PdlClient (selv om det kanskje bare er en i første omgang.

Bør kanskje wrappe det som caches sammen med et timestamp slik at ikke levetiden på cachen blir like lang som pod'ens levetid (feks bør ikke cachede objekter som er mer enn 24 timer gamle brukes).

Reply:
Jeg innså en mulig feil med den cachen --> Dersom pdl faktisk ikke er oppdatert med aktiv ident, så er det dumt at vi cacher den før den sjekken. Hvis det faktisk ikke er oppdatert vil dette da skje:
1. Henter identer fra pdl 
2. Lagrer disse i cachen
3. Gjør sjekken om den nye identene vi fikk på kafka faktisk er den gjeldende identen nå
4. Steg 3 kaster en feil, og appen gjør en retry for å lese recorden på nytt etter 60 sekunder
5. Når vi nå skal hente identer fra pdl så brukes jo bare den ikke oppdaterte verdien som er i cachen, og det vil fortsette å feile

Grunnen til at jeg lagde cache var for å ikke måtte gjøre kallet 4 ganger, men nå fant jeg en annen måte å gjøre det på, selv om den kanskje ikke er like pen 🤷🏼‍♂️ Se siste commit.